### DISCUSSION TITLE
Automatically generating sitemaps

### DISCUSSION THREAD

**Discussion by samrobbins85** [vercel/next.js#17566] [Ideas]
There are approaches for creating sitemaps, such as https://leerob.io/blog/nextjs-sitemap-robots and https://www.npmjs.com/package/nextjs-sitemap-generator . But it would be nice if there was a solution built into Next.js to provide a sitemap as I can't think of many websites that would be worse for having a sitemap enabled. If someone didn't want a sitemap it could be configured in `next.config.js`

The inspiration for this is that [Hugo](https://gohugo.io/) takes this approach

**Comment by leerob** [comment]
I agree! I'd love to see this built-in. First, it'd probably be helpful to consolidate all of the possible workarounds and use cases that exist and then create an RFC with a proposed configuration or API. 

Is this something you'd be interested in?

**Reply by samrobbins85** [reply]
Yeah I'd love to help! I'll start work on that soon

**Reply by leerob** [reply]
Awesome, thank you so much 🙏  This will be very helpful.

**Reply by leerob** [reply]
Also, maybe @IlusionDev can help chime in on which pieces should be included in the framework and which shouldn't.

**Reply by IlusionDev** [reply]
@leerob Of course I can help, make it a native feature will be awesome.
